addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/discovery/NetworkDiscoveryService.java

```diff
@@ -23,25 +28,47 @@
 import org.eclipse.smarthome.core.thing.ThingUID;
 import org.openhab.binding.network.service.DiscoveryCallback;
 import org.openhab.binding.network.service.NetworkUtils;
+import org.osgi.service.component.ComponentContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * The {@link NetworkDiscoveryService} is responsible for discovering devices on
  * the current Network. It uses every Network Interface which is connected to a network.
+ * It tries common TCP ports to connect to, ICMP pings and ARP pings.
  *
+ * @author David Graeff
  * @author Marc Mettke - Initial contribution
  */
 public class NetworkDiscoveryService extends AbstractDiscoveryService implements DiscoveryCallback {
+    static final int PING_TIMEOUT_IN_MS = 500;
+    // TCP port 548 (Apple Filing Protocol (AFP))
+    // TCP port 554 (Windows share / Linux samba)
+    // TCP port 1025 (XBox / MS-RPC)

```

Comment:
If you generate the same thingUID it should not be duplicated but the last one should 'win'